scaleway/resource_object_bucket.go

```diff
@@ -431,6 +440,31 @@ func resourceScalewayObjectBucketRead(ctx context.Context, d *schema.ResourceDat
 	_ = d.Set("name", bucketName)
 	_ = d.Set("region", region)
 
+	// Get object_lock_enabled
+	objectLockConfiguration, err := s3Client.GetObjectLockConfigurationWithContext(ctx, &s3.GetObjectLockConfigurationInput{
+		Bucket: aws.String(bucketName),
+	})
+	if err != nil {
+		s3err, isS3Err := err.(awserr.Error)
+
+		if isS3Err {
+			switch s3err.Code() {
+			case s3.ErrCodeNoSuchBucket:
+				tflog.Error(ctx, fmt.Sprintf("Bucket %q was not found - removing from state!", bucketName))
+				d.SetId("")
+				return nil
+			case "ObjectLockConfigurationNotFoundError":
+				_ = d.Set("object_lock_enabled", false)
+			default:
+				return diag.FromErr(fmt.Errorf("couldn't read bucket: %s", err))
+			}
+		} else {
+			return diag.FromErr(fmt.Errorf("couldn't read bucket: %s", err))
```

Comment:
use `isS3Error` and define `"ObjectLockConfigurationNotFoundError"` in a constant in `errors_object.go`